pkg/transaction/transaction.go

```diff
@@ -283,13 +283,15 @@ func (t *transactionService) prepareTransaction(ctx context.Context, request *Tx
 		return nil, ErrGasPriceTooLow
 	}
 
-	return types.NewTx(&types.LegacyTx{
-		Nonce:    nonce,
-		To:       request.To,
-		Value:    request.Value,
-		Gas:      gasLimit,
-		GasPrice: gasPrice,
-		Data:     request.Data,
+	return types.NewTx(&types.DynamicFeeTx{
+		ChainID:   t.chainID,
+		Nonce:     nonce,
+		To:        request.To,
+		Value:     request.Value,
+		Gas:       gasLimit,
+		GasTipCap: gasPrice,
```

Comment:
What is the TipCap and FeeCap stand for? Are you able to do test the transactions?

I would even suggest having a separate go program which just sends some transactions in the new format and check if it works. You can connect to testnet/mainnet vpn and run your program with the same RPC endpoint as the node and try to issue some txs.

Reply:
Yes, the test can be simple as deploying a custom image to dev-bee-cluster for example.

Reply:
@istae @aloknerurkar 
https://goerli.etherscan.io/tx/0x84ef13070d4f5a651bed53196c2f0a51288f8d61a84eb39e6ea231ae951d9b1d

You can find the london tx using above link. You can verify it by going to attributes.

@ralph-pichler @agazso @notanatol @janos  for kind notice

Reply:
@aloknerurkar so TipCap is max priority fee per gas we are willing to pay to a miner to prioritize our transaction where as fee is max fee per gas we are willing to pay per transaction.

reference: https://www.liquity.org/blog/how-eip-1559-changes-the-transaction-fees-of-ethereum